packages/core/scripts/local/getGasEthHistoricalPrice.js

```diff
@@ -0,0 +1,110 @@
+/**
+ * @notice This is an example script demonstrating how to get a historical median gas price at a specific timestamp.
+ * It could be used for example to query a GASETH-TWAP-1Mx1M price before committing a vote for a DVM price request.
+ *
+ * @dev Prequisites: before running, you will need to set a GOOGLE_APPLICATIONS_CREDENTIALS environment variable for a service account.
+ * @dev This service account will need GCP admin or BigQuery permissions. This guide provides further instructions: https://cloud.google.com/docs/authentication/getting-started
+ * @dev How to run: yarn truffle exec ./scripts/local/getGasEthHistoricalPrice.js --network mainnet_mnemonic --time 1612137600 --identifier GASETH-TWAP-1Mx1M
+ */
+const { BigQuery } = require("@google-cloud/bigquery");
+const highland = require("highland");
+const moment = require("moment");
+const { fromWei } = web3.utils;
+const argv = require("minimist")(process.argv.slice(), { string: ["identifier", "time"] });
+const { createQuery } = require("./GasEthQuery");
+const BigNumber = require("bignumber.js");
+
+const client = new BigQuery();
+const dateFormat = "YYYY-MM-DD HH:mm:ss";
+const precisionToUse = 18;
+
+async function getGasEthHistoricalPrice(callback) {
+  try {
+    // If user did not specify an identifier, provide a default value.
+    let queryIdentifier;
+    if (!argv.identifier) {
+      queryIdentifier = "GASETH-TWAP-1Mx1M";
+      console.log(`Optional '--identifier' flag not specified, defaulting to: ${queryIdentifier}`);
+    } else {
+      queryIdentifier = argv.identifier;
+    }
+
+    // Function to get the current time.
+    const getTime = () => Math.round(new Date().getTime() / 1000);
+
+    // If user specified a timestamp, then use it, otherwise default to the current time.
+    let queryTime;
+    if (!argv.time) {
+      queryTime = getTime();
+      console.log(
+        `Optional '--time' flag not specified, defaulting to the current Unix timestamp (in seconds): ${queryTime}`
+      );
+    } else {
+      queryTime = argv.time;
+    }
```

Comment:
Is there ever a situation where someone would want to query with the current time and not a specified timestamp?

Reply:
Yeah for sanity checking or a quick run for example